--- urcl-os/urclos3.urcl	2024-06-08 06:07:28.702767834 +0000
+++ urcl-os/urclos3_del.urcl	2024-06-08 06:04:33.252767750 +0000
@@ -1499,16 +1499,36 @@
 .Scal_Gets_Loop
 IN R1 %TEXT
 BRE .Scal_Gets_End R1 R6
-BRZ ~+2 R7
+BRE .Scal_Gets_Del R1 127
+BRZ .Scal_Gets_EchoSkip R7
 OUT %TEXT R1
+.Scal_Gets_EchoSkip
 STR R3 R1
 INC R3 R3
 DEC R2 R2
+.Scal_Gets_DelSkip
 BNZ .Scal_Gets_Loop R2
 .Scal_Gets_End
 STR R3 R0
 RET
 
+.Scal_Gets_Del
+BRE .Scal_Gets_DelNone R2 R3
+DEC R3 R3
+INC R2 R2
+OUT %TEXT 0x1b
+OUT %TEXT '['
+OUT %TEXT '1'
+OUT %TEXT 'D'
+OUT %TEXT 0x1b
+OUT %TEXT '['
+OUT %TEXT 'K'
+JMP .Scal_Gets_DelSkip
+
+.Scal_Gets_DelNone
+OUT %TEXT 7
+JMP .Scal_Gets_DelSkip
+
 .Scal_Getc //TODO: make non blocking
 ADD R3 R3 .VirtualRegisters
 IN R1 %TEXT
